src/views/Dashboard/Customer/Home/GeneticData/Add.vue

```diff
@@ -0,0 +1,179 @@
+<template lang="pug">
+  .genetic-data-add
+    .genetic-data-add__wrapper
+      .genetic-data-add__title {{ isEdit ? "Edit Genetic Data" : "Add Genetic Data"}} 
+      .genetic-data-add__forms
+        ui-debio-input(
+          :rules="titleRule"
+          v-model="document.title"
+          variant="small"
+          label="Title"
+          placeholder="Add title"
+          outlined 
+          block 
+          validate-on-blur
+        )
+
+        ui-debio-textarea(
+          :rules="descriptionRule"
+          v-model="document.description"
+          variant="small"
+          label="Description"
+          placeholder="Add Description"
+          outlined 
+          block 
+          validate-on-blur
+        )
+
+        ui-debio-file(
+          v-model="document.file"
+          variant="small"
+          :rules="fileRule"
+          :accept="['.pdf', '.vcf', '.vcg.gz']"
+          label="Upload File"
+          placeholder="Choose File"
+          validate-on-blur
+        )
+        .genetic-data-add__tx-weight
+          span.genetic-data-add__tx-weight-text Estimated transaction weight 
+            v-tooltip.visible(right)
+              template(v-slot:activator="{ on, attrs }")
+                v-icon.dialog-confirmation__trans-weight-icon(
+                  style="font-size: 12px;"
+                  color="primary"
+                  dark
+                  v-bind="attrs"
+                  v-on="on"
+                ) mdi-alert-circle-outline 
+              span(style="font-size: 10px;") Total fee paid in DBIO to execute this transaction.
+
+          span( style="font-size: 12px;" ) 0.0014 DBIO
+          
+        Button(
+          :disabled="!disable"
+          block
+          color="secondary"
+          @click="onSubmit"
+        ) Submit
+
+      SuccessDialog(
+        :show="isSuccess"
+        title="Success"
+        @close="isSuccess = false"
+      )
+
+
+</template>
+
+<script>
+import Button from "@/common/components/Button"
+import errorMessage from "@/common/constants/error-messages"
+import { validateForms } from "@/common/lib/validate"
+import { checkCircleIcon } from "@/common/icons"
+import SuccessDialog from "@/common/components/Dialog/SuccessDialog"
+
+export default {
+  name: "AddGeneticData",
+  
+  components: { Button, SuccessDialog },
+
+  mixins: [validateForms],
+
+  data: () => ({
+    document: {
+      title: "",
+      description: "",
+      file: null
+    },
+    isEdit: false,
+    isSuccess: false,
+    checkCircleIcon
+  }),
+
+  computed: {
+    disable() {
+      const { title, description, file } = this.document
+      return !title, description, file
+    },
+
+    englishAlphabeth() {
+      return  val => (val && /^[A-Za-z0-9!@#$%^&*\\(\\)\-_=+:;"',.\\/? ]+$/.test(val)) || errorMessage.INPUT_CHARACTER("English alphabet")
+    },
+
+    fieldRequired() {
+      return val => !!val || errorMessage.REQUIRED
+    },
```

Comment:
Move rules like this to a `common` folder, make it accessible to the other pages.

Reply:
done